### DISCUSSION TITLE
Show the full path in popup

### DISCUSSION THREAD

**Discussion by Louis-udm** [microsoft/pylance-release#2704] [Enhancement]
Hi,

Why not show in the title the full path in the popup?
When review a long code, we often need to know where the object comes from.
<img width="526" alt="Screen Shot 2021-08-19 at 4 12 37 PM" src="https://user-images.githubusercontent.com/25377679/130137552-7a975b3b-40fc-4402-adc6-97e9599546b3.png">

**Comment by jakebailey** [comment]
Can you clarify what you mean?

If you mean to show `dd.DataFrame` or similar, note that modules often re-export from other modules, so the "real" full path to a particular thing may be unexpected.

**Comment by Louis-udm** [comment]
> Can you clarify what you mean?
> 
> If you mean to show `dd.DataFrame` or similar, note that modules often re-export from other modules, so the "real" full path to a particular thing may be unexpected.

Thx for the reply. Let me show you what PyCharm did:
<img width="500" alt="Screen Shot 2021-08-19 at 4 45 27 PM" src="https://user-images.githubusercontent.com/25377679/130141600-d003ca0a-08f9-40b8-b3c2-075675fd246f.png">

With a full path in the title of popup, I can immediately know what is the object(also class/function) of DataFrame from, Pandas or Dask.

**Comment by Louis-udm** [comment]
What I mean is the definition path.

**Comment by erictraut** [comment]
Thanks for the feedback. We've been told by other users that the full definition path is quite verbose and unnecessary. Based on this feedback (and other similar input), we have been looking for ways to be more concise in our hover text and error messages.

Out of curiosity, why do you find the full definition path to be useful? Do you find that there are name collisions in your code, so the full path is needed to disambiguate between same-named symbols? For example, in the screen shot above, is there ambiguity about what type `DataFrame` refers to? Or is there some other benefit that you get from viewing the full definition path? If we better understand the value you get from seeing the full path, we might be able to come up with other ways to provide that value without making everything more verbose.

**Comment by Louis-udm** [comment]
I don't think those who think the full definition path is verbose are representative of everyone. I can understand that for small projects, or for developers who only focus on their own code, maybe who don't need a full definition path. But for larger projects (with many packages), and for those who often need to review other people's PRs, a definition path in a popup is very convenient, otherwise I often need to right-click to bring up the definition function.

Regarding the examples I gave, about the name collision, actually the code that developer already open the PR does not have a name collision, but a collision in our brain, and reviewers who are not familiar with the code often need to clarify where the object comes from. So why does Pycharm give definition path in popup?

I think it would be a good solution to at least provide an option in the setting.

Thx!


> Thanks for the feedback. We've been told by other users that the full definition path is quite verbose and unnecessary. Based on this feedback (and other similar input), we have been looking for ways to be more concise in our hover text and error messages.
> 
> Out of curiosity, why do you find the full definition path to be useful? Do you find that there are name collisions in your code, so the full path is needed to disambiguate between same-named symbols? For example, in the screen shot above, is there ambiguity about what type `DataFrame` refers to? Or is there some other benefit that you get from viewing the full definition path? If we better understand the value you get from seeing the full path, we might be able to come up with other ways to provide that value without making everything more verbose.

**Comment by diogo-rossi** [comment]
> I think it would be a good solution to at least provide an option in the setting.

I agree. It should at least provide the option in the settings.

To illustrate: Jedi already shows the definition path ("Full name").
I also don't think it is verbose. It is very useful.

See bellow the comparison when using Pylance and Jedi as language server.

### Jedi:
![image](https://user-images.githubusercontent.com/57204023/136576898-f4288c07-2a23-48b2-b716-fb47b948cd97.png)

### Pylance:
![image](https://user-images.githubusercontent.com/57204023/136577329-a32ef275-8d23-4f0a-93ac-bd20e62c3060.png)

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by diogo-rossi** [comment]
I think the "thumbs up" should be converted to "up votes" when moving the discussion to enhancement request.

I'm still wanting this feature, but my vote wasn't count.
